comm/src/main/scala/coop/rchain/comm/transport/StreamObservable.scala

```diff
@@ -0,0 +1,36 @@
+package coop.rchain.comm.transport
+
+import cats._, cats.data._, cats.implicits._
+import coop.rchain.shared.Log
+import coop.rchain.comm.PeerNode
+import monix.eval.Task
+import monix.execution.{Cancelable, Scheduler}
+import monix.reactive.Observable
+import monix.reactive.observers.Subscriber
+import monix.reactive.OverflowStrategy._
+import monix.reactive.subjects.ConcurrentSubject
+
+case class ToStream(peerNode: PeerNode, blob: Blob)
+
+class StreamObservable(bufferSize: Int)(implicit scheduler: Scheduler)
+    extends Observable[ToStream] {
+
+  val subject = ConcurrentSubject.publishToOne[ToStream](DropNew(bufferSize))
+
+  def stream(peers: List[PeerNode], blob: Blob): Task[Unit] =
+    peers
+      .traverse(
+        peer =>
+          Task.fromFuture {
+            subject.onNext(ToStream(peer, blob))
+          }
+      )
+      .as(())
+
+  def unsafeSubscribeFn(subscriber: Subscriber[ToStream]): Cancelable = {
```

Comment:
The point of the names starting with unsafe is to trigger an alarm signal in the developer writing that code and make him think twice whether he should use that function in that context or not. 

We have a LOT of calls to unsafe* functions and it became a habit to deal with unsafe stuff. Can we use a SafeSubscriber here?

Reply:
@odeac this is the monix API I has to stick with

Reply:
what's a safesubscriber? :)

Reply:
I was looking at this: https://monix.io/docs/2x/reactive/observers.html#safe-subscriber  I'm not sure how it actually works but I'm worried about our familiarity with the unsafe* stuff in our code

Reply:
after the call I understand the problems and why unsafeSubscribeFn is unsafe and needed